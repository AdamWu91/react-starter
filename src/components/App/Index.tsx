import React from "react";
import logo from "./logo.svg";
import "@components/App/App.scss";

const App: React.FC = () => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>
        Edit <code>src/App.tsx</code> and save to reload.
      </p>
      <a className="App-link" href="#reactjs.org" rel="noopener noreferrer">
        Learn React
      </a>
    </header>
  </div>
);
export default App;
