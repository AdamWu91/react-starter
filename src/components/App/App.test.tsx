import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

import App from '@components/App/Index';

test('renders learn react link', () => {
  const result = shallow(<App />).contains('Learn React');
  expect(result).toBeTruthy();
});

test('check app snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
