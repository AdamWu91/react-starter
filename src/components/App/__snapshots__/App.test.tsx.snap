// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check app snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="#reactjs.org"
        rel="noopener noreferrer"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;
